PolyEngine/Engine/Src/InputSystem.cpp

```diff
@@ -7,74 +7,35 @@ using namespace Poly;
 
 void InputSystem::InputPhase(World* world)
 {
-	InputWorldComponent* com = world->GetInputWorldComponent();
+	InputWorldComponent com = world->GetInputWorldComponent();
 
-	for (int i = 0; i < 256; i++)
-		com->PrevKey[i] = com->CurrKey[i];
+	memcpy(&com.PrevKey, &com.CurrKey, static_cast<int>(eKey::_COUNT) * sizeof(bool));
```

Comment:
You can use operator=() that is part of EnumArray. ;) Right now, this is undefined behaviour, you're relaying strictly on the fact that Data is the first member of EnumArray.